packages/lib/src/interactions/SelectionTool.tsx

```diff
@@ -46,69 +62,81 @@ function SelectionTool(props: Props) {
 
       const { target, pointerId } = sourceEvent;
       (target as Element).setPointerCapture(pointerId);
-      setStartPoint(worldToData(unprojectedPoint));
+      setSelection({
+        startPoint: worldToData(unprojectedPoint),
+        endPoint: worldToData(unprojectedPoint),
+      });
+      setOriginPoint(new Vector2(unprojectedPoint.x, unprojectedPoint.y));
 
       if (onSelectionStart) {
         onSelectionStart();
       }
     },
-    [onSelectionStart, shouldInteract, worldToData]
+    [onSelectionStart, setSelection, shouldInteract, worldToData]
   );
 
   const onPointerMove = useCallback(
     (evt: CanvasEvent<PointerEvent>) => {
-      if (!startPoint) {
+      if (!originPoint) {
         return;
       }
 
       const { unprojectedPoint, sourceEvent } = evt;
-      const point = worldToData(boundPointToFOV(unprojectedPoint, camera));
-      setEndPoint(point);
+      const worldEndPoint = new Vector2(unprojectedPoint.x, unprojectedPoint.y);
+      const worldStartPoint = getWorldStartPoint(originPoint, worldEndPoint);
+      const selection = {
+        startPoint: worldToData(boundPointToFOV(worldStartPoint, camera)),
+        endPoint: worldToData(boundPointToFOV(worldEndPoint, camera)),
```

Comment:
The `boundPointToFOV` must now be applied to both `startPoint` and `endPoint`. As it works in world coordinates, it is easier to deduce the `startPoint` from `originPoint` (termed `worldStartPoint`) in the world space, which is why I store `originPoint` in world space.